Dockerize showtime-plus for PCPD-P

kjcioffi opened the pull request.

kjcioffi committed:
Add DEBUG check so toolbar is disabled not imported in production.

kjcioffi committed:
Add functional Dockerfile.

kjcioffi committed:
Add compose file.

ryaustin commented on Dockerfile:
Let's add a step to migrate before running the server. 
It's easy to overlook but is a common cause of startup issues.
```
# Run migrations and start the application
CMD python manage.py migrate --noinput && gunicorn -b 0.0.0.0:80 -b 0.0.0.0:8000 core.wsgi
```

ryaustin reviewed (changes requested):
Looking good here Kevin, showtime is definitly dockerized - it's running perfectly on my system.

In dev, I had to make some key changes - you may consider adding Dockerfile.dev so that anyone can run the project in docker if they wish.
1. in your compose.yaml I replaced the image key with `build: .`
2. in Dockerfile I changed build dependencies to use both dependencies and added migrations. 

I am requesting those changes:
- [x] Because the dockerfile is not available in the repo or on a container hub, remove the image key and replace it with `build: .`
- [x] Add environment variables in the compose.yaml or add a .env-template in git repo
- [x] Set the build dependencies to dev, or both or adjust. I needed both dev and prod dependencies to run the project.
- [x] Add a readme for how to runt he project in docker (and test the steps).

kjcioffi committed:
Update README.md for containerization. Additionally, clarified some things concerning getting the app running locally.

kjcioffi committed:
Add dockerignore file.

kjcioffi committed:
Remove pip-tools from dev dependencies.

kjcioffi committed:
Add environment variables and arguments to Dockerfile.
   - No longer need to add the .env file to the Docker image.

kjcioffi committed:
Compose file to use environment variables, arguments, and builds the application.

kjcioffi committed:
.env template file for convenience.

kjcioffi committed:
Added more context on how to run the project in a container.

kjcioffi commented:
> Looking good here Kevin, showtime is definitly dockerized - it's running perfectly on my system.
> 
> 
> 
> In dev, I had to make some key changes - you may consider adding Dockerfile.dev so that anyone can run the project in docker if they wish.
> 
> 1. in your compose.yaml I replaced the image key with `build: .`
> 
> 2. in Dockerfile I changed build dependencies to use both dependencies and added migrations. 
> 
> 
> 
> I am requesting those changes:
> 
> - [x] Because the dockerfile is not available in the repo or on a container hub, remove the image key and replace it with `build: .`
> 
> - [x] Add environment variables in the compose.yaml or add a .env-template in git repo
> 
> - [x] Set the build dependencies to dev, or both or adjust. I needed both dev and prod dependencies to run the project.
> 
> - [x] Add a readme for how to runt he project in docker (and test the steps).

Thanks, Ryan. I appreciate the suggestions!

Here are some keys changes from my side.

1. The application is now built with Docker compose when called.

2. The environment variables are now fed into the image without having to copy the .env into it.

3. README.md has been updated.

I was facing some issues with building the project and running it like a development environment. Building the container with gunicorn is much more straightforward and will look to run a container with manage.py at a later time.
